cloudferrylib/os/image/glance_image.py

```diff
@@ -227,8 +227,8 @@ def delete_image(self, image_id):
         self.glance_client.images.update(image_id, protected=False)
         self.glance_client.images.delete(image_id)
 
-    def get_image_by_id(self, image_id):
-        for glance_image in self.get_image_list():
+    def get_image_by_id(self, image_id, is_filtering=True):
+        for glance_image in self.get_image_list(is_filtering):
```

Comment:
This is to avoid unexpected errors, as had already transparent filtering previously realized